ebean-ddl-generator/src/main/java/io/ebeaninternal/dbmigration/model/ModelDiff.java

```diff
@@ -125,6 +140,21 @@ public void compareTo(ModelContainer newModel) {
     // search for tables that are no longer used
     for (MTable existingTable : baseModel.getTables().values()) {
       if (!newTables.containsKey(existingTable.getName())) {
+        // Compounded foreign keys used in many-to-many mapping tables
+        existingTable.getCompoundKeys().forEach(it -> addAlterForeignKey(it.dropForeignKey(existingTable.getName())));
+        // Foreign keys through direct mapping (oneToMany)
+        existingTable.allColumns().stream()
```

Comment:
```suggestion
        existingTable.allColumns().stream()
        .filter(col -> col.getForeignKeyIndex() != null)
        .forEach(col -> {
            AlterForeignKey fk = new AlterForeignKey();
            fk.setName(col.getForeignKeyName());
            fk.setIndexName(col.getForeignKeyIndex());
            fk.setColumnNames(DdlHelp.DROP_FOREIGN_KEY);
            fk.setTableName(existingTable.getName());
            addAlterForeignKey(fk);
        });
```
Just for better reading